extensions/cornerstone-dicom-pmap/src/getSopClassHandlerModule.ts

```diff
@@ -0,0 +1,224 @@
+      return displaySet.referencedVolumeId;
+    }
+
+    const referencedDisplaySet = displaySet.getReferenceDisplaySet();
+    const referencedVolumeURI = referencedDisplaySet.displaySetInstanceUID;
+    const referencedVolumeId = `${VOLUME_LOADER_SCHEME}:${referencedVolumeURI}`;
+
+    displaySet.referencedVolumeURI = referencedVolumeURI;
+    displaySet.referencedVolumeId = referencedVolumeId;
+
+    return referencedVolumeId;
+  };
+
+  displaySet.load = async ({ headers }) =>
+    await _load(displaySet, servicesManager, extensionManager, headers);
+
+  return [displaySet];
+}
+
+async function _load(
+  displaySet,
+  servicesManager: AppTypes.ServicesManager,
+  extensionManager,
+  headers
+) {
+  const volumeId = `${VOLUME_LOADER_SCHEME}:${displaySet.displaySetInstanceUID}`;
+  const volumeLoadObject = cache.getVolumeLoadObject(volumeId);
+
+  if (volumeLoadObject) {
+    return volumeLoadObject.promise;
+  }
+
+  displaySet.loadStatus.loading = true;
+
+  // We don't want to fire multiple loads, so we'll wait for the first to finish
+  // and also return the same promise to any other callers.
+  // loadPromises[SOPInstanceUID] = new Promise(async (resolve, reject) => {
+  const promise = _loadParametricMap({
+    extensionManager,
+    servicesManager,
+    displaySet,
+    headers,
+  });
+
+  cache.putVolumeLoadObject(volumeId, { promise }).catch(err => {
+    throw err;
+  });
+
+  promise
+    .then(() => {
+      displaySet.loadStatus.loading = false;
+      displaySet.loadStatus.loaded = true;
+    })
+    .catch(err => {
+      displaySet.loadStatus.loading = false;
+      throw err;
+    });
+
+  return promise;
+}
+
+async function _loadParametricMap({
+  extensionManager,
+  servicesManager,
+  displaySet,
+  headers,
+}: withAppTypes) {
+  const utilityModule = extensionManager.getModuleEntry(
+    '@ohif/extension-cornerstone.utilityModule.common'
+  );
+
+  const { dicomLoaderService } = utilityModule.exports;
+  const arrayBuffer = await dicomLoaderService.findDicomDataPromise(displaySet, null, headers);
+  const referencedVolumeId = displaySet.getReferencedVolumeId();
+  const cachedReferencedVolume = cache.getVolume(referencedVolumeId);
+
+  if (!cachedReferencedVolume) {
+    throw new Error(
+      'Referenced Volume is missing for the SEG, and stack viewport SEG is not supported yet'
+    );
+  }
+
+  const { imageIds } = cachedReferencedVolume;
+  const results = await adaptersPMAP.Cornerstone3D.ParametricMap.generateToolState(
+    imageIds,
+    arrayBuffer,
+    metaData
+  );
+  const { pixelData } = results;
+  const TypedArrayConstructor = pixelData.constructor;
+  const paramMapId = displaySet.displaySetInstanceUID;
+
+  const derivedVolume = await volumeLoader.createAndCacheDerivedVolume(referencedVolumeId, {
+    volumeId: paramMapId,
+    targetBuffer: {
+      type: TypedArrayConstructor.name, // 'Int16Array'
+    },
+  });
+
+  derivedVolume.getScalarData().set(pixelData);
```

Comment:
I believe the `createAndCacheDerivedVolume` method can accept `pixelData` as an argument, eliminating the need to handle it separately afterwards. Take a look at the parameters for `createAndCacheDerived` to confirm this.